core/tests/test-dependencies/route-to-module-mapping-generator.ts

```diff
@@ -0,0 +1,441 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview This script is used to generate an Angular route to module mapping.
+ */
+
+import path from 'path';
+import fs from 'fs';
+import {Route} from '@angular/router';
+import {
+  CallExpression,
+  ObjectLiteralExpression,
+  SourceFile,
+  ts,
+} from 'ts-morph';
+import {
+  project,
+  ROOT_DIRECTORY,
+  getDecoratorNodesByTextFromSourceFile,
+  resolveModuleRelativeToRoot,
+  AngularDecorators,
+  getValueFromLiteralStringOrBinaryExpression,
+} from './typescript-ast-utilities';
+
+// List of routing module file paths.
+const ROUTING_MODULE_FILE_PATHS = [
+  path.resolve(
+    ROOT_DIRECTORY,
+    'core/templates/pages/oppia-root/routing/app.routing.module.ts'
+  ),
+  path.resolve(
+    ROOT_DIRECTORY,
+    'core/templates/pages/lightweight-oppia-root/routing/app.routing.module.ts'
+  ),
+];
+
+// List of routes that are not defined in routing modules. These routes are
+// manually mapped to their corresponding modules. These routes are currently not
+// covered by the routing module scraping logic since they are loaded by webpack and
+// not definied in the above routing modules.
```

Comment:
Please mention that this list cannot be added to.

Reply:
Done.